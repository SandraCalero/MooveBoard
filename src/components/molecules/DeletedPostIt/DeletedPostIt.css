.restoreAndClose {
	width: 100%;
	padding: 0.8rem 0.8rem 0;
	display: flex;
	align-items: flex-start;
	justify-content: space-between;
}
