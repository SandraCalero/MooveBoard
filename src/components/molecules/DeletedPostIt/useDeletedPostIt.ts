import { IPostItWithEvent } from 'globals/definitions/postItProps';
import { useDispatch } from 'react-redux';
import { restorePostIt } from 'redux/slices/trashspace';

export const useDeletedPostIt = ({
	id,
	content,
	shouldOpenModal,
}: IPostItWithEvent) => {
	const dispatch = useDispatch();

	const handleOnContextMenu = () => {
		console.log('Click derecho');
	};

	const handleRestore = () => dispatch(restorePostIt({ id, content }));

	const handleClose = () => shouldOpenModal({ id, content });

	return {
		handleRestore,
		handleOnContextMenu,
		handleClose,
	};
};

export default useDeletedPostIt;
