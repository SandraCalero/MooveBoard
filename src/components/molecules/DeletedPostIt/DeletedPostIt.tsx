import Button from 'components/atoms/Button/Button';
import { IPostItWithEvent } from 'globals/definitions/postItProps';
import restorePostIt from 'assets/icons/restorePostIt.png';
import useDeletedPostIt from './useDeletedPostIt';

export default function DeletedPostIt({
	id,
	content,
	showModal,
}: IPostItWithEvent) {
	const { hideContextMenu, handleRestore, handleClose } = useDeletedPostIt({
		id,
		content,
		showModal,
	});

	return (
		<div
			className='flex flex-col items-center justify-center bg-yellow
			shadow-xl w-max dark:bg-cerulean-500 dark:shadow-lg
			dark:shadow-cerulean-700 transition-colors dark:border
			dark:border-grey dark:text-mine-shaft-600'
			id={id.toString()}
			onClick={hideContextMenu}
			aria-hidden='true'
		>
			<div className='flex justify-between w-full px-2 pt-2 items-start'>
				<Button
					variant='h-7 w-7 my-0 mx-0 opacity-20 hover:opacity-100
					relative hover:after:content-["Restore"] after:absolute
					after:left-8 after:bg-white after:rounded-lg after:p-1
					after:text-sm dark:text-mine-shaft-900 dark:opacity-40
					dark:hover:opacity-100 transition-colors'
					onClick={handleRestore}
					icon={restorePostIt}
				/>
				<Button
					variant='h-6 w-6 my-0 mx-0 opacity-20 text-xl
					hover:opacity-100 dark:text-mine-shaft-900 dark:opacity-40
					dark:hover:opacity-100 transition-colors'
					onClick={handleClose}
					text='X'
				/>
			</div>
			<div className='break-words p-2 w-52 basis-52 bg-transparent'>
				{content}
			</div>
		</div>
	);
}
