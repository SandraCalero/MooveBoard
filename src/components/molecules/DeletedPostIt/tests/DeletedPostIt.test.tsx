import { render } from 'utils/tests/test-utils';
import DeletedPostIt from '../DeletedPostIt';

describe('<DeletedPostIt />', () => {
	it('Snapshot', () => {
		const showModal = jest.fn();
		const { container, getByText } = render(
			<DeletedPostIt
				id={1}
				content='Testing Deleted Post It'
				showModal={showModal}
			/>
		);
		expect(container).toMatchSnapshot();
		expect(getByText('Testing Deleted Post It')).toBeTruthy();
	});
});
