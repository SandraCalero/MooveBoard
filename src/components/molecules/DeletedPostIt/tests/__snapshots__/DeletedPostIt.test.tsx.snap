// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DeletedPostIt /> should match the snapshot 1`] = `
<div>
  <div
    aria-hidden="true"
    class="flex flex-col items-center justify-center bg-yellow shadow-xl w-max dark:bg-cerulean-500 dark:shadow-xl dark:shadow-cerulean-700 transition-colors dark:border dark:border-grey dark:text-mine-shaft-600"
    id="1"
  >
    <div
      class="flex justify-between w-full px-2 pt-2 items-start"
    >
      <button
        class="flex flex-col items-center justify-center border-none
			m-2 p-0	bg-transparent hover:opacity-90 active:translate-y-1 h-7 w-7 my-0 mx-0 opacity-20 hover:opacity-100 relative hover:after:content-[\\"Restore\\"] after:absolute after:left-8 after:bg-white after:rounded-lg after:p-1 after:text-sm dark:text-mine-shaft-900 dark:opacity-40 dark:hover:opacity-100 transition-colors"
        name="button"
        type="button"
      >
        <img
          alt=""
          class="h-4/5 w-4/5"
          src="restorePostIt.png"
        />
      </button>
      <button
        class="flex flex-col items-center justify-center border-none
			m-2 p-0	bg-transparent hover:opacity-90 active:translate-y-1 h-6 w-6 my-0 mx-0 opacity-20 text-xl hover:opacity-100 dark:text-mine-shaft-900 dark:opacity-40 dark:hover:opacity-100 transition-colors"
        name="button"
        type="button"
      >
        X
      </button>
    </div>
    <div
      class="break-words p-2 w-52 basis-52  bg-transparent"
    >
      Testing Deleted Post It
    </div>
  </div>
</div>
`;
